dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp

```diff
@@ -113,21 +113,13 @@ void PageDirectoryFactory::loadEdit(const PageDirectoryPtr & dir, const PageEntr
             max_applied_ver = r.version;
         max_applied_page_id = std::max(r.page_id, max_applied_page_id);
 
-        // We can not avoid page id from being reused under some corner situation. Try to do gcInMemEntries
-        // and apply again to resolve the error.
-        if (bool ok = applyRecord(dir, r, /*throw_on_error*/ false); unlikely(!ok))
-        {
-            dir->gcInMemEntries();
-            applyRecord(dir, r, /*throw_on_error*/ true);
-            LOG_FMT_INFO(DB::Logger::get("PageDirectoryFactory"), "resolve from error status done, continue to restore");
-        }
+        applyRecord(dir, r);
```

Comment:
I think the test case `PageDirectoryGCTest.RestoreWithDuplicateID` is failed after we removed this retry logic?